mirgecom/diffusion.py

```diff
@@ -49,24 +53,102 @@ def _u_flux(discr, alpha, q_tpair):
     return discr.project(q_tpair.dd, "all_faces", flux_weak)
 
 
-def diffusion_operator(discr, alpha, u):
+class DiffusionBoundary(metaclass=abc.ABCMeta):
+    """
+    Diffusion boundary base class.
+
+    .. automethod:: get_q_flux
+    .. automethod:: get_u_flux
+    """
+
+    @abc.abstractmethod
+    def get_q_flux(self, discr, alpha, dd, u):
+        """Compute the flux for *q* on the boundary corresponding to *dd*."""
+        pass
+
+    def get_u_flux(self, discr, alpha, dd, q):
+        """Compute the flux for *u* on the boundary corresponding to *dd*."""
+        pass
+
+
+class DirichletDiffusionBoundary(DiffusionBoundary):
+    """
+    Dirichlet boundary condition for the diffusion operator.
+
+    .. automethod:: __init__
+    """
+
+    def __init__(self, value=0.):
```

Comment:
```suggestion
    def __init__(self, value):
```
[PEP20](https://www.python.org/dev/peps/pep-0020/): Explicit is better than implicit.